src/states/Game.gd

```diff
@@ -37,6 +40,19 @@ func _input(event: InputEvent) -> void:
 		pause_menu.open_menu()
 
 
+func _process(delta: float) -> void:
+	if time_remaining > 0:
+		time_remaining -= delta
+		get_node("UI/CountdownTimer").text = "Time Remaining: %d" % floor(time_remaining)
+	else: # time_remaining <= 0
+		if get_tree().is_network_server():
+			rpc("end_of_match")
+			end_of_match()
+		elif not get_tree().get_network_peer():
+			var error := get_tree().change_scene("res://src/states/Menu.tscn")
+			assert(not error)
```

Comment:
Is it even worth having the timer in free play?

Reply:
Good point - in free play, the player might just want to practice with no time limit